app/controllers/application_controller.rb

```diff
@@ -177,21 +177,28 @@ def error_handler(e)
 
     logger.fatal("Error caught: [#{e.class.name}] #{e.message}\n#{e.backtrace.join("\n")}")
 
-    msg = case e
-          when ::ActionController::RoutingError
-            _("Action not implemented")
-          when ::AbstractController::ActionNotFound # Prevent Rails showing all known controller actions
-            _("Unknown Action")
-          else
-            e.message
-          end
+    if(e.class == MiqException::RbacPrivilegeException)
```

Comment:
Had to use `e.class` since `e` was received as a string message - `'The user is not authorized for this task or item'`

Reply:
`e` shouldn't be a string, otherwise line 178 should blow up on the `e.message` call. Let me know how I can help test/debug it.

Reply:
flyby tangent:. We probably want to use `kind_of`:

```suggestion
    if(e.kind_of?(MiqException::RbacPrivilegeException)
```

also curious why we can't just throw this into the when
(do we need a redirect?)

Reply:
yes, I initially created a new case for `MiqException::RbacPrivilegeException` and handled the redirect inside the format.html.
And It didn't break at line #178. let me try again.